### ISSUE TITLE
Installing CLI on Arch Linux with Snap

### ISSUE THREAD

**Issue by gbbirkisson** [enonic/cli-enonic#141]
When installing the cli with `sudo snap install enonic` on arch linux it has problems:
* It does not seem to like my java home: JAVA_HOME is set to an invalid directory
* It is unable to start service on port 8080
* The ~/.enonic folder does not appear but is is located at ~/snap/enonic/5/.enonic

When installing with `sudo snap install --devmode enonic`:
* Still complains about java home
* It is able to start the service on port 8080

When downloading the binary manually from the enonic repo and installing it has no problems.

The reason could be because goreleaser is releasing [the snap in strict mode](https://github.com/enonic/cli-enonic/blob/acf3357a236829c73bb0929a3c082df3efdbabc4/.goreleaser.yml#L253).

I am wondering if adding plugs to the `.goreleaser.yml` file would do the trick? Something like this:
```yaml
snapcraft:
  ...
  name: enonic
  ...
  confinement: strict

  apps:
    enonic:
      plugs: ["network", "network-bind", "personal-files", "system-files"]

  plugs:
    personal-files:
      read:
      - $HOME/.enonic
      write:
      - $HOME/.enonic
    system-files:
      read:
      - $JAVA_HOME
```

Links:
* [Snapcraft confinement](https://snapcraft.io/docs/reference/confinement)
* [Snapcraft supported interfaces](https://forum.snapcraft.io/t/supported-interfaces/7744)
* [Plugs in goreleaser](https://goreleaser.com/snapcraft/)

**Comment by sigdestad** [comment]
So, things might still be working fine with Ubuntu?

**Comment by pmi** [comment]
@sigdestad Just checked again :) It does work on the amazon ubuntu box
@gbbirkisson Sad it didn't work out of the box, I will add plugs to the next version ! Let me know if that helps

**Comment by pmi** [comment]
Created an interface request https://forum.snapcraft.io/t/request-for-auto-connection-for-enonic/11733

**Comment by gbbirkisson** [comment]
Here are the results of my latest testing:

### TL;DR
* Snap installation works fine
* Creation of sandboxes works fine.
* Starting sandboxes works fine, but just because environment variable `JAVA_HOME` is ignored.
* Creation of projects works.
* Creation/Updating of `myproject/.enonic` works.
* Project deployment does **NOT** work.
    * Unless with some hacks 

### Start clean

```bash
$ sudo rm -r $HOME/.enonic && sudo snap refresh enonic && enonic -v
snap "enonic" has no updates available
Enonic CLI version 1.0.11
```

### Create new sandbox

```bash
$ enonic sandbox create Sandbox1
? Enonic XP distribution: linux-sdk-7.0.2
Downloading distro [======================================================] 100.00%
Unzipping distro...Done

Sandbox 'Sandbox1' created with distro 'linux-sdk-7.0.2'.
```

### Start Sandbox1

```bash
$ enonic sandbox start Sandbox1
                         _____
____________________________(_)______   ____  _________
_  _ \_  __ \  __ \_  __ \_  /_  ___/   __  |/_/__  __ \
/  __/  / / / /_/ /  / / /  / / /__     __>  < __  /_/ /
\___//_/ /_/\____//_/ /_//_/  \___/     /_/|_| _  .___/
                                               /_/

# Enonic XP 7.0.2
# Built on 2019-07-01T11:05:25+0200 (hash = e094d90068a0ffc0dd48bf69b3aa60c986bc7aea, branch = 7.0)
# OpenJDK 64-Bit Server VM 11.0.3 (AdoptOpenJDK)
# Linux 4.19.62-1-MANJARO (amd64)
# Install directory is /home/gummi/.enonic/distributions/enonic-xp-linux-sdk-7.0.2

.... Standard logs ....

2019-08-06 14:23:09,337 INFO  c.e.x.l.i.framework.FrameworkService - Started Enonic XP in 4438 ms
```

**Note:** Changing `JAVA_HOME` here has no effect because `$HOME/.enonic/distributions/enonic-xp-linux-sdk-7.0.2/bin/setenv.sh` always sets `JAVA_HOME=$HOME/.enonic/distributions/enonic-xp-linux-sdk-7.0.2/jdk/`. So even running `JAVA_HOME=NOT_VALID_DIR enonic sandbox start Sandbox1` works. This is probably correct behavior.

### Create new project

```bash
$ enonic project create

Loading starters from enonic market...Done.
? Starter Vanilla Starter (Clean and basic project structures)
? Project name com.example.myproject
? Destination folder myproject
? Project version 1.0.0-SNAPSHOT

Initializing project...
Enumerating objects: 61, done.
Counting objects: 100% (61/61), done.
Compressing objects: 100% (35/35), done.
Total 194 (delta 27), reused 46 (delta 20), pack-reused 133

? A sandbox is required for your project, select one: Sandbox1 ( linux-sdk-7.0.2 )

Project created in '/home/gummi/tests/myproject' and linked to 'Sandbox1'

? Open Vanilla Starter docs in the browser ? No
Vanilla Starter docs: https://github.com/enonic/starter-vanilla
```

### Testing project sandbox config

```bash
$ cd myproject
$ enonic sandbox create Sandbox2
? Enonic XP distribution: linux-sdk-7.0.2

Sandbox 'Sandbox2' created with distro 'linux-sdk-7.0.2'.

$ enonic project sandbox Sandbox2 && \
      cat .enonic && \
      enonic project sandbox Sandbox1 && \
      cat .enonic

Sandbox 'Sandbox2' set as default.
sandbox = "Sandbox2"

Sandbox 'Sandbox1' set as default.
sandbox = "Sandbox1"
```

### Deploying projects

```bash
$ enonic project deploy

Deploying to sandbox 'Sandbox1'...

ERROR: JAVA_HOME is set to an invalid directory: /usr/lib/jvm/default

Please set the JAVA_HOME variable in your environment to match the
location of your Java installation.
```

This error is thrown by the projects `./gradlew` script. So I added to the top of the script:

```bash
#!/usr/bin/env sh
echo "enonic-cli command: ./gradlew $@"
.
.
.
```

Now if I run the command again I can get the command that the CLI is trying to run:

```bash
$ enonic project deploy

Deploying to sandbox 'Sandbox1'...
enonic-cli command: ./gradlew deploy -Dorg.gradle.java.home=/home/gummi/.enonic/distributions/enonic-xp-linux-sdk-7.0.2/jdk -Dxp.home=/home/gummi/.enonic/sandboxes/Sandbox1/home

ERROR: JAVA_HOME is set to an invalid directory: /usr/lib/jvm/default

Please set the JAVA_HOME variable in your environment to match the
location of your Java installation.
```

Run the command manually that the CLI runs:

```bash
$ ./gradlew deploy -Dorg.gradle.java.home=/home/gummi/.enonic/distributions/enonic-xp-linux-sdk-7.0.2/jdk -Dxp.home=/home/gummi/.enonic/sandboxes/Sandbox1/home
.
.
.
BUILD SUCCESSFUL in 1s
4 actionable tasks: 4 executed
```

So it seems that in the Snapcraft is blocking access to  `/usr/lib/jvm/default`.  As a matter of fact it blocks all attempts to access any JDK I have installed `/usr/lib/jvm/*`

### Deploying projects (Bundled SDK)

It also works if I set the SDK with the one bundled with the distribution:

```bash
$ JAVA_HOME=$HOME/.enonic/distributions/enonic-xp-linux-sdk-7.0.2/jdk enonic project deploy

Deploying to sandbox 'Sandbox1'...

BUILD SUCCESSFUL in 0s
4 actionable tasks: 4 up-to-date
```

**Comment by gbbirkisson** [comment]
This issue is now fixed for me. @anatol-sialitski does everything work for you?
